enzyme/Enzyme/GradientUtils.h

```diff
@@ -934,7 +934,16 @@ class GradientUtils : public CacheUtility {
     if (auto arg = dyn_cast<Argument>(ptr)) {
       assert(arg->getParent() == oldFunc);
     }
-    ptr = invertPointerM(ptr, BuilderM);
```

Comment:
Can you change this to instead check if BuilderM is a forward pass block or not to choose whether to do the lookup?

Reply:
done